src/lib/comms/sol-connman-impl-connman.c

```diff
@@ -899,5 +899,8 @@ sol_connman_del_service_monitor(
 SOL_API int
 sol_connman_get_service_vector(struct sol_vector *vector)
 {
+    vector = &_ctx.service_vector;
+    SOL_NULL_CHECK(vector, -EINVAL);
```

Comment:
This is wrong.
for two reasons:
First it should be `struct sol_vector **vector`
So you would do something like: `*vector = &_ctx.service_vector;`

You also need to invert the order here, NULL_CHECK must be placed before the attribution